yt/frontends/art/fields.py

```diff
@@ -141,53 +141,42 @@ def _metallicity(field, data):
         atoms = ['C', 'N', 'O', 'F', 'Ne', 'Na', 'Mg', \
         'Al', 'Si', 'P', 'S', 'Cl', 'Ar', 'K', 'Ca', 'Sc', \
         'Ti', 'V', 'Cr', 'Mn', 'Fe', 'Co', 'Ni', 'Cu', 'Zn']
-        for atom in atoms:
+        def _specific_metal_density_function(atom):
```

Comment:
This seems problematic to have two nested functions.  Is this a mistake?

Reply:
No this is actually the major fix. Before, the _specific_metal_density function would use the local variable `atom`. But since it was not evaluating until later, when it _did_ get evaluated, `atom` meant `'Zn'`, no matter which field you were querying.

Now the function `_specific_metal_density_function` creates a separate field definition function for each atom, and that will locally know the right atom to use.